xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -650,13 +655,20 @@ void onData(ParsedResource<T> parsedResource, String version, long updateTime) {
         resourceDeletionIgnored = false;
       }
       if (!Objects.equals(oldData, data)) {
-        for (ResourceWatcher<T> watcher : watchers) {
-          notifyWatcher(watcher, data);
+        for (ResourceWatcher<T> watcher : watchers.keySet()) {
```

Comment:
Consider using `entrySet()`, so you don't have to look up the value. Ditto below.